modules/katello_sync.py

```diff
@@ -107,115 +104,59 @@
   until: async_job_result.finished
   retries: 999
   delay: 10
-
 '''
 
-RETURN = '''# '''
+RETURN = ''' # '''
 
 try:
-    from nailgun import entities, entity_fields, entity_mixins
-    from nailgun.config import ServerConfig
-    has_import_error = False
-except ImportError as e:
-    has_import_error = True
-    import_error_msg = str(e)
-
-from ansible.module_utils.basic import AnsibleModule
-
-
-class NailGun(object):
-
-    def __init__(self, server, entities, module):
-        self._server = server
-        self._entities = entities
-        self._module = module
-        entity_mixins.TASK_TIMEOUT = 180000  # Publishes can sometimes take a long, long time
-
-    def find_organization(self, name):
-        org = self._entities.Organization(self._server, name=name)
-        response = org.search(set(), {'search': 'name="{}"'.format(name)})
-
-        if len(response) == 1:
-            return response[0]
-        else:
-            self._module.fail_json(msg="No organization found for %s" % name)
-
-    def find_product(self, name, organization):
-        org = self.find_organization(organization)
-
-        product = self._entities.Product(self._server, name=name, organization=org)
-        response = product.search()
-
-        if len(response) == 1:
-            return response[0]
-        else:
-            self._module.fail_json(msg="No Product found for %s" % name)
-
-    def find_repository(self, name, product, organization):
-        product = self.find_product(product, organization)
-
-        repository = self._entities.Repository(self._server, name=name, product=product)
-        repository._fields['organization'] = entity_fields.OneToOneField(entities.Organization)
-        repository.organization = product.organization
-        response = repository.search()
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        ping_server,
+        find_organization,
+        find_product,
+        find_repository,
+    )
 
-        if len(response) == 1:
-            return response[0]
-        else:
-            self._module.fail_json(msg="No Repository found for %s" % name)
+    from nailgun import entity_mixins
+    entity_mixins.TASK_TIMEOUT = 180000  # Publishes can sometimes take a long, long time
```

Comment:
Is it worth doing this depending on the value of `synchronous`?

Reply:
It would be nice to have it at sync time. But setting it globally in the module, i would keep it directly after the import.